src/component/basecomponent.go

```diff
@@ -0,0 +1,39 @@
+package component
+
+import (
+	// "fmt"
+	// "encoding/json"
+	// "strings"
+	"strconv"
+)
+
+/*BaseComponent object definition */
+type BaseComponent struct {
+	UID     int
+	Name    string
+	Enabled bool
+	OnState bool
+}
+
+//Enable - make this component available to command
+func (c *BaseComponent) Enable(restoreState bool) {
+	c.Enabled = true
+}
+
+//Disable - force this component to an off or safe state and make it unavaible to command
+func (c *BaseComponent) Disable() {
+	c.Enabled = false
+	c.OnState = false
+}
+
+/*CurrentStateSting just for pretty printing the device info */
+func (c *BaseComponent) CurrentStateSting() string {
```

Comment:
Typo. But also maybe lets follow the structure used in solenoid.go and implement String() on each object for pretty printing